src/fastalloc/iter.rs

```diff
@@ -0,0 +1,339 @@
+            OperandConstraintKind::FixedReg != op.constraint().into()
+            && OperandConstraintKind::Reuse != op.constraint().into()
+            && op.pos() == OperandPos::Late
+        )
+    }
+
+    pub fn non_reuse_late_def(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::Reuse != op.constraint().into()
+            && op.pos() == OperandPos::Late
+            && op.kind() == OperandKind::Def
+        )
+    }
+
+    pub fn non_fixed_non_reuse_early(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::FixedReg != op.constraint().into()
+            && OperandConstraintKind::Reuse != op.constraint().into()
+            && op.pos() == OperandPos::Early
+        )
+    }
+
+    pub fn reuse(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op| OperandConstraintKind::Reuse == op.constraint().into())
+    }
+
+    pub fn non_reuse_early_def(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::Reuse != op.constraint().into()
+            && op.pos() == OperandPos::Early
+            && op.kind() == OperandKind::Def
+        )
+    }
+
+    pub fn fixed_early(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::FixedReg == op.constraint().into()
+            && op.pos() == OperandPos::Early
+        )
+    }
+
+    pub fn fixed_late(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::FixedReg == op.constraint().into()
+            && op.pos() == OperandPos::Late
+        )
+    }
+
+    pub fn non_reuse_def(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::Reuse != op.constraint().into()
+            && op.kind() == OperandKind::Def
+        )
+    }
+
+    pub fn non_fixed_non_reuse_late_use(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::FixedReg != op.constraint().into()
+            && OperandConstraintKind::Reuse != op.constraint().into()
+            && op.pos() == OperandPos::Late
+            && op.kind() == OperandKind::Use
+        )
+    }
+
+    pub fn non_fixed_non_reuse_late_def(&self) -> impl Iterator<Item = (usize, Operand)> + 'a {
+        self.matches(|op|
+            OperandConstraintKind::FixedReg != op.constraint().into()
+            && OperandConstraintKind::Reuse != op.constraint().into()
+            && op.pos() == OperandPos::Late
+            && op.kind() == OperandKind::Def
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use alloc::vec::Vec;
+    use alloc::vec;
+    use crate::{PReg, RegClass};
+    use super::*;
+
+    // Using a new function because Operand::new isn't a const function
```

Comment:
If you change `debug_assert_eq!` to `debug_assert!` then it will work in a const function. We lose the debug print of the values but that isn't a big deal.